package/base-files/files/lib/functions/system.sh

```diff
@@ -1,31 +1,36 @@
 # Copyright (C) 2006-2013 OpenWrt.org
 
 . /lib/functions.sh
-. /usr/share/libubox/jshn.sh
 
 get_mac_binary() {
 	local path="$1"
-	local offset="$2"
+	local offset="${2:-0}"
+	local length="${3:-6}"
+	local mac
 
 	if ! [ -e "$path" ]; then
 		echo "get_mac_binary: file $path not found!" >&2
 		return
 	fi
 
-	hexdump -v -n 6 -s $offset -e '5/1 "%02x:" 1/1 "%02x"' $path 2>/dev/null
+	mac=$(head -c $((offset + length)) "$path" | tail -c "$length" |
+		hexdump -e '6/1 "%02x"' 2>/dev/null)
```

Comment:
This replaces a single fork/exec with one subshell and two fork/exec to invoke external `head` and `tail` instead of utilizing hexdump's builtin skipping.